### ISSUE TITLE
Error while running background job: An exception occurred while executing after update to 0.3.6

### ISSUE THREAD

**Issue by RealRancor** [owncloud/music#395]
### Steps to reproduce
1. Have a working music 0.3.5 installation
2. Update to music 0.3.6
3. Have a look at the owncloud.log

Probably related to the newer SQL statements from https://github.com/owncloud/music/pull/347

### Expected behaviour
No messages about an error while running background job shown in logfiles

### Actual behaviour
Messages about an error while running background job shown in logfiles

### Server configuration
**Operating system**: Debian Jessie

**Web server:** NginX 1.7.x

**Database:** MariaDB 10.0.x

**PHP version:** 5.6.x

**ownCloud version:** 8.1.0

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

#### ownCloud log (data/owncloud.log)
```
{"reqId":"Mwb7z+Y2+joMQFLa4nL+","remoteAddr":"","app":"no app in context","message":"Error while running background job: An exception occurred while executing 'UPDATE `oc_music_albums` SET `cover_file_id` = NULL WHERE `cover_file_id` IS NOT NULL AND `cover_file_id` IN (SELECT `cover_file_id` FROM `oc_music_albums` LEFT JOIN `oc_filecache` ON `cover_file_id`=`fileid` WHERE `fileid` IS NULL);':\n\nSQLSTATE[HY000]: General error: 1093 Table 'oc_music_albums' is specified twice, both as a target for 'UPDATE' and as a separate source for data","level":3,"time":"2015-07-10T08:00:19+02:00"}
```

**Comment by MorrisJobke** [comment]
I started to reproduce this issue with integration tests over here: https://github.com/owncloud/music/pull/396

**Comment by MorrisJobke** [comment]
https://travis-ci.org/owncloud/music/builds/70364648

MySQL fails with:

```
1) OCA\Music\Utility\HelperTest::testCleanup

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'UPDATE `oc_music_albums` SET `cover_file_id` = NULL WHERE `cover_file_id` IS NOT NULL AND `cover_file_id` IN (SELECT `cover_file_id` FROM `oc_music_albums` LEFT JOIN `oc_filecache` ON `cover_file_id`=`fileid` WHERE `fileid` IS NULL);':

SQLSTATE[HY000]: General error: 1093 You can't specify target table 'oc_music_albums' for update in FROM clause
```

:white_check_mark: I can reproduce. Now I can fix it :)

**Comment by RealRancor** [comment]
Nice :+1: Wasn't sure if this is maybe only MariaDB related.

**Comment by MorrisJobke** [comment]
There is most likely also a fix: https://github.com/owncloud/music/pull/372 ...currently travis is verifying if this fixes the issue :) Let the machines do the hard work :beers:

**Comment by MorrisJobke** [comment]
Fixed with #372
